src/components/20-molecules/file-upload/index.js

```diff
@@ -164,37 +172,63 @@ class AXAFileUpload extends LitElement {
   }
 
   handleFileDeletion(index) {
-    let clonedFiles = [];
+    let clonedOriginalFiles = [];
+    let clonedCompressedFiles = [];
 
-    this.allDroppedFiles = this.files.length + this.faultyFiles.length - 1;
+    this.numberOfDroppedFiles = this.files.length + this.faultyFiles.length - 1;
+
+    const allOriginalFiles = this.validOriginalFiles.concat(
+      this.faultyOriginalFiles
+    );
 
     if (index >= this.files.length) {
-      // faulty file
-      clonedFiles = [...this.faultyFiles];
-      clonedFiles.splice(index - this.files.length, 1);
-      this.faultyFiles = clonedFiles;
+      // remove a faulty file from file arrays
+      clonedOriginalFiles = [...this.faultyOriginalFiles];
+      clonedOriginalFiles.splice(index - this.validOriginalFiles.length, 1);
+      this.faultyOriginalFiles = clonedOriginalFiles;
+
+      clonedCompressedFiles = [...this.faultyCompressedFiles];
+      clonedCompressedFiles.splice(index - this.validCompressedFiles.length, 1);
+      this.faultyCompressedFiles = clonedCompressedFiles;
     } else {
-      // valid file
-      clonedFiles = [...this.files];
-      clonedFiles.splice(index, 1);
-      this.files = clonedFiles;
-      if (this.files.length + 1 === this.maxNumberOfFiles) {
+      // remove a valid file from file arrays
+      clonedOriginalFiles = [...this.validOriginalFiles];
+      clonedOriginalFiles.splice(index, 1);
+      this.validOriginalFiles = clonedOriginalFiles;
+
+      clonedCompressedFiles = [...this.validCompressedFiles];
+      clonedCompressedFiles.splice(index, 1);
+      this.validCompressedFiles = clonedCompressedFiles;
```

Comment:
...and use this new function for these 2 blocks here

Reply:
done